components/dashboard/src/service/public-api.ts

```diff
@@ -120,3 +131,70 @@ export function projectToProtocol(project: Project): ProtocolProject {
         },
     };
 }
+
+let user: User | undefined;
+export function updateUser(newUser: User | undefined) {
+    user = newUser;
+}
+
+function createServiceClient<T extends ServiceType>(type: T, jsonRpcClient?: PromiseClient<T>): PromiseClient<T> {
+    return new Proxy(createPromiseClient(type, transport), {
+        get(grpcClient, prop) {
+            // TODO(ak) remove after migration
+            async function resolveClient(): Promise<PromiseClient<T>> {
+                if (!jsonRpcClient) {
+                    return grpcClient;
+                }
+                const isEnabled = await getExperimentsClient().getValueAsync("dashboard_public_api_enabled", false, {
+                    // TODO(ak): is not going to work for getLoggedInUser itself
```

Comment:
IMO we should pass the host as well like we do [here](https://github.com/gitpod-io/gitpod/blob/cf7d0e950087966ac7fc209ca5e8c7f0fb640df6/components/dashboard/src/data/featureflag-query.ts#L47). That way we can turn it on a per-instance level.